Source/displayinfo/include/displayinfo.h

```diff
@@ -0,0 +1,170 @@
+    DISPLAYINFO_HDCP_1X,
+    DISPLAYINFO_HDCP_2X,
+    DISPLAYINFO_HDCP_UNKNOWN
+} displayinfo_hdcp_protection_t;
+
+typedef enum displayinfo_error_type {
+    ERROR_NONE = 0,
+    ERROR_UNKNOWN = 1,
+    ERROR_INVALID_INSTANCE = 2,
+} displayinfo_error_t;
+
+/**
+* \brief Will be called if there are changes regaring the display output, you need to query 
+*        yourself what exacally is changed
+*
+* \param session The session the notification applies to.
+* \param userData Pointer passed along when \ref displayinfo_register was issued.
+*/
+typedef void (*displayinfo_updated_cb)(struct displayinfo_type* session, void* userdata);
+
+/**
+ * \brief Get a implementation instance name to use for displayinfo_instance based on index\
+ *        If \ref length is 0  and \ref buffer is NULL, only a check is assumed.
+ * 
+ * \param index The index of a DisplayInfo implementation
+ * \param length Length \ref buffer, can be 0 for a check only
+ * \param buffer Pointer to a buffer, can be NULL if \ref length is 0 for a check only
+ * 
+ * \return true if a instance was found for the index, false otherwise.
+ **/
+bool displayinfo_enumerate(const uint8_t index, const uint8_t length, char* buffer);
+
+/**
+ * \brief Get a \ref displayinfo_type instance that matches the a DisplayInfo implementation.
+ *
+ * \param displayName Name a the implementation that holds the.
+ * 
+ * \return \ref displayinfo_type instance, NULL on error.
+ **/
+struct displayinfo_type* displayinfo_instance(const char displayName[]);
+
+/**
+ * \brief Release the \ref displayinfo_type instance.
+ * 
+ * \param instance Instance of \ref displayinfo_type.
+ * 
+ **/
+void displayinfo_release(struct displayinfo_type* instance);
+
+/**
+ * \brief Register for updates of the display output.
+ * 
+ * \param instance Instance of \ref displayinfo_type.
+ * \param callback Callback that needs to be called if a chaged is deteced.
+ * \param userdata The user data to be passed back to the \ref displayinfo_updated_cb callback.
+ * 
+ **/
+void displayinfo_register(struct displayinfo_type* instance, displayinfo_updated_cb callback, void* userdata);
+
+/**
+ * \brief Unregister for updates of the display output.
+ * 
+ * \param instance Instance of \ref displayinfo_type.
+ * \param callback Callback that was used to \ref displayinfo_registet
+ * 
+ **/
+void displayinfo_unregister(struct displayinfo_type* instance, displayinfo_updated_cb callback);
+
+/**
+ * \brief Checks if a audio passthrough is enabled.
+ * 
+ * \param instance Instance of \ref displayinfo_type.
+ * 
+ * \return true if audio passthrough is enabled, false otherwise.
+ **/
+bool displayinfo_is_audio_passthrough(struct displayinfo_type* instance);
+
+/**
+ * \brief Checks if a display is connected to the display output.
+ * 
+ * \param instance Instance of \ref displayinfo_type.
+ * 
+ * \return true a dispplay is connected, false otherwise.
+ * 
+ **/
+bool displayinfo_connected(struct displayinfo_type* instance);
```

Comment:
If we want to make this header file also valid C, I don't think bool is allowed

Reply:
I was not sure myself as well so did a bit of googling:
Standard C (since C99) provides a boolean type, called _Bool . By including the header stdbool. h , one can use the more intuitive name bool and the constants true and false . ... Objective-C also has a separate Boolean data type BOOL , with possible values being YES or NO , equivalents of true and false respectively.
So I guess it os oke if we include <stdbool.h>

Reply:
And we do so I guess... GTG...